src/index.ts

```diff
@@ -1,128 +1,156 @@
+    publicKey: string;
+  };
+  uploadOptions: {
+    useUniqueFileName?: boolean;
+    tags?: string[];
+    folder?: string;
+  };
+  enableDatedFolders: boolean;
 }
 
 class Store extends StorageBase {
-    private urlEndpoint: string;
-    private privateKey: string;
-    private publicKey: string;
-    private uploadOptions: ImageKitStoreConfig["uploadOptions"]
-    private _imagekit: ImageKit
-
-    constructor(config: ImageKitStoreConfig) {
-        // @ts-ignore
-        super(config)
-
-        const {
-            auth,
-            uploadOptions = {}
-        } = config
-
-        const {
-            urlEndpoint = process.env.IMAGEKIT_STORE_URL_ENDPOINT,
-            privateKey = process.env.IMAGEKIT_STORE_PRIVATE_KEY,
-            publicKey = process.env.IMAGEKIT_STORE_PUBLIC_KEY
-        } = auth || {};
-
-        const {
-            useUniqueFileName = true,
-            tags = [],
-            folder = '/'
-        } = uploadOptions;
-
-        this.urlEndpoint = urlEndpoint ?? "";
-        this.privateKey = privateKey ?? "";
-        this.publicKey = publicKey ?? "";
-        this.uploadOptions = {
-            useUniqueFileName: (useUniqueFileName.toString() === "false" || useUniqueFileName === false) ? false : true,
-            tags,
-            folder: folder
-        };
-
-        this._imagekit = new ImageKit({
-            urlEndpoint: this.urlEndpoint,
-            publicKey: this.publicKey,
-            privateKey: this.privateKey
-        })
-    }
-
-    async exists(fileName: string, targetDir?: string): Promise<boolean> {
-        console.log("exists", fileName, targetDir)
-        const filePath = stripLeadingSlash(join(targetDir || "", fileName));
-        const response = await got(new URL(filePath, this.urlEndpoint), {
-            responseType: 'buffer',
-            resolveBodyOnly: true
-        })
-        if (response) return true
-        return false;
-    }
-
-    async save(image: StorageBase.Image, targetDir?: string): Promise<string> {
-        console.log("save", image.name)
-        const folder = targetDir || this.getTargetDir(this.uploadOptions.folder)
-        const fileName = this.getSanitizedFileName(image.name);
-
-        const file = await readFileAsync(image.path);
-
-        const uploadResponse = await this._imagekit.upload({
-            file,
-            fileName,
-            useUniqueFileName: this.uploadOptions.useUniqueFileName,
-            folder,
-            tags: this.uploadOptions.tags
-        })
-
-        var result = new URL(uploadResponse.url)
-        result.searchParams.append("updatedAt", new Date().getTime().toString())
-        return result.toString()
+  private urlEndpoint: string;
+  private privateKey: string;
+  private publicKey: string;
+  private enableDatedFolders: boolean;
+  private uploadOptions: ImageKitStoreConfig["uploadOptions"];
+  private _imagekit: ImageKit;
+
+  constructor(config: ImageKitStoreConfig) {
+    // @ts-ignore
+    super(config);
```

Comment:
Do we need to send config to `super()` because StorageBase doesn't need config in constructor for [StorageBase](https://github.com/TryGhost/Ghost-Storage-Base/blob/main/BaseStorage.js)

Reply:
done